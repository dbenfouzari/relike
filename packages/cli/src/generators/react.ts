import { paramCase, snakeCase } from "change-case";
import fs from "fs";
import path from "path";

import { replace, replaceFileName } from "../utils/template";

/**
 * Generates component index file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateIndexFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateIndexFile = (key: string, componentName: string) => {
  const templateIndexPath = path.resolve(__dirname, "../templates/index/index.ts.txt");

  const templateIndex = fs.readFileSync(templateIndexPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateIndexPath, key, componentName)),
    content: replace(templateIndex, key, componentName, paramCase),
  };
};

/**
 * Generates component style file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStyleFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStyleFile = (key: string, componentName: string) => {
  const templateStylePath = path.resolve(__dirname, "../templates/react-style/__name__.module.scss.txt");

  const templateStyle = fs.readFileSync(templateStylePath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStylePath, key, componentName)),
    content: replace(templateStyle, key, componentName, snakeCase),
  };
};

/**
 * Generates component story file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStoryFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStoryFile = (key: string, componentName: string) => {
  const templateStoryPath = path.resolve(__dirname, "../templates/react-story/__name__.stories.tsx.txt");

  const templateStory = fs.readFileSync(templateStoryPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStoryPath, key, componentName)),
    content: replace(templateStory, key, componentName),
  };
};

/**
 * Generates component test file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateTestFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateTestFile = (key: string, componentName: string) => {
  const templateTestPath = path.resolve(__dirname, "../templates/react-test/__name__.test.tsx.txt");

  const templateTest = fs.readFileSync(templateTestPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateTestPath, key, componentName)),
    content: replace(templateTest, key, componentName),
  };
};

/**
 * Generates component file
 *
 * @param   {string}  key           The key to replace
 * @param   {string}  componentName The component name
 * @param   {boolean} hasStyle      Does the component have style
 * @example
 * generateComponentFile("name", name, flags.style);
 * @returns {any}                   File path and content.
 */
export const generateComponentFile = (key: string, componentName: string, hasStyle: boolean) => {
  const templateComponentPath = hasStyle
    ? path.resolve(__dirname, "../templates/react-component/__name__(if:style).tsx.txt")
    : path.resolve(__dirname, "../templates/react-component/__name__.tsx.txt");

  const templateComponent = fs.readFileSync(templateComponentPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateComponentPath, key, componentName)),
    content: replace(templateComponent, key, componentName),
  };
};
