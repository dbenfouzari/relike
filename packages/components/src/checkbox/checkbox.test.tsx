import { act, fireEvent, render } from "@testing-library/react";

import Checkbox from "./checkbox";

describe("Checkbox", () => {
  it("should render successfully", () => {
    const { container } = render(<Checkbox />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render successfully when default checked", () => {
    const { container } = render(<Checkbox defaultChecked />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should handle click", () => {
    const { getByTestId } = render(<Checkbox defaultChecked />);

    const input = getByTestId("input");

    act(() => {
      fireEvent.click(input);
    });
  });
});
