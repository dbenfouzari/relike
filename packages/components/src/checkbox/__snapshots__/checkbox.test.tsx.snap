// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render successfully 1`] = `
<label
  class="wrapper"
>
  <div
    class="outer"
  >
    <input
      class="checkbox checkbox-0-2-1 checkbox-d0-0-2-2"
      data-testid="input"
      type="checkbox"
    />
    <span
      class="checkmark"
    >
      <span
        class="material-icons sc-bdnxRM bYKVnq"
      >
        check
      </span>
    </span>
  </div>
</label>
`;

exports[`Checkbox should render successfully when default checked 1`] = `
<label
  class="wrapper"
>
  <div
    class="outer"
  >
    <input
      checked=""
      class="checkbox checkbox-0-2-1 checkbox-d1-0-2-3"
      data-testid="input"
      type="checkbox"
    />
    <span
      class="checkmark checkmark__checked"
    >
      <span
        class="material-icons sc-bdnxRM bYKVnq"
      >
        check
      </span>
    </span>
  </div>
</label>
`;
